core/kazoo_auth/src/kz_auth_keys.erl

```diff
@@ -308,3 +314,60 @@ gen_private_key() ->
 erlint(MPInts) when is_list(MPInts) -> [erlint(X) || X <- MPInts ];
 erlint(<<Size:32, Int:Size/unit:8>>) -> Int.
 
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% Reset Kazoo private key, first get KeyId by from config.
+%% * Check if the document exists, if not create it,
+%% * generate a new private key and put it in cache
+%% @end
+%%--------------------------------------------------------------------
+-spec reset_kazoo_private_key() -> 'ok' | {'error', any()}.
+reset_kazoo_private_key() ->
+    lager:debug("trying to reset kazoo private key"),
+    reset_kazoo_private_key(kz_auth_apps:get_auth_app(<<"kazoo">>)).
+
+-spec reset_kazoo_private_key(map() | ne_binary()) -> 'ok' | {'error', any()}.
+reset_kazoo_private_key(#{pvt_server_key := KeyId}) ->
+    reset_kazoo_private_key(KeyId);
+reset_kazoo_private_key(#{}) ->
+    {'error', 'invalid_identity_provider'};
+reset_kazoo_private_key(?NE_BINARY=KeyId) ->
+    lager:debug("check if key document (kid ~s) exists", [KeyId]),
+    case kz_datamgr:open_cache_doc(?KZ_AUTH_DB, KeyId) of
+        {'ok', JObj} -> reset_kazoo_private_key(KeyId, JObj);
+        {'error', 'not_found'} ->
+            lager:debug("private key document (kid ~s) is not exists, creating it...", [KeyId]),
+            case new_private_key(KeyId) of
+                {ok, PrivateKey} ->
+                    lager:debug("private key document (kid ~s) is created successfully", [KeyId]),
+                    PublicKey = get_public_key_from_private_key(PrivateKey),
+                    store(KeyId, PublicKey);
+                {'error', _Reason}=Error ->
+                    lager:debug("failed to create new private key (kid ~s): ~p", [KeyId, _Reason]),
+                    Error
+            end;
+        {'error', _Reason}=Error ->
+            lager:debug("failed to open private key document (kid ~s)", [KeyId, _Reason]),
+            Error
+    end.
+
+-spec reset_kazoo_private_key(ne_binary(), kz_json:object()) -> 'ok' | {'error', any()}.
+reset_kazoo_private_key(KeyId, JObj) ->
+    lager:debug("resetting kazoo private key ~s", [KeyId]),
+    {'ok', PrivateKey} = gen_private_key(),
+    Options = [{'doc_type', <<"system_key">>}
+              ,{'rev', kz_doc:revision(JObj)}
+              ,{'content_type', ?SYSTEM_KEY_ATTACHMENT_CTYPE}
+              ],
+    case kz_datamgr:put_attachment(?KZ_AUTH_DB, KeyId, ?SYSTEM_KEY_ATTACHMENT_NAME, to_pem(PrivateKey), Options) of
+        {'ok', _} ->
+            lager:debug("refreshing private/public cache for key ~s", [KeyId]),
+
+            store({'private', KeyId}, PrivateKey),
+            PublicKey = get_public_key_from_private_key(PrivateKey),
+            store(KeyId, PublicKey);
+        {'error', _Err}=Err ->
+            lager:debug("error ~p saving generated system key ~s", [_Err, KeyId]),
+            Err
+    end.
```

Comment:
the above code is duplication of existing code. the `reset` can be achieved by setting `?SYSTEM_KEY_ID` to a new rand_hex